sys/net/application_layer/dns/dns.c

```diff
@@ -216,14 +188,43 @@ int sock_dns_query(const char *domain_name, void *addr_out, int family)
     }
 
     for (int i = 0; i < SOCK_DNS_RETRIES; i++) {
+        uint8_t *buf = dns_buf;
+
+        sock_dns_hdr_t *hdr = (sock_dns_hdr_t*) buf;
+        memset(hdr, 0, sizeof(*hdr));
+        hdr->id = 0; /* random? */
```

Comment:
I know this is just pulled down, but this should at least come from a variable now, that it is re-set on every retry.

Reply:
`random_bytes(&hdr->id, sizeof(hdr->id));`?